File: Inheritance/Inheritance(Java)/Single_Level.java
class Rectangle
{
    float length;
    float breadth;

    Rectangle()
    {
        length = 5;
        breadth = 6;
    }
}

class Area extends Rectangle
{
    public float calculate()
    {
        return length * breadth;
    }
}

class Single_Level
{
    public static void main(String args[])
    {
        Area a = new Area();

        System.out.println("Area = " + a.calculate() + " square meters");
    }
}

/* Output

Area = 30 square meters

*/

File: Inheritance/Inheritance(C++)/Single_Level.cpp
#include <iostream>

using namespace std;

class Rectangle
{
    protected:
       float length, breadth;

   public:
       Rectangle(): length(0.0), breadth(0.0)
        {
            length = 5;
            breadth = 6;
        }
};

/* Area class is derived from base class Rectangle. */
class Area : public Rectangle
{
    public:
        float calc()
        {
            return length * breadth;
        }
};

int main()
{
     Area a;
     cout << "Area = " << a.calc() << " square meters" << endl;

     return 0;
}


/* OUTPUT

Area = 30 square meter

*/
